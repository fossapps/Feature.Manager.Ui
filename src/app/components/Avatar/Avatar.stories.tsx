import React from "react";
import { Avatar } from "./Avatar";

export default {
  component: Avatar,
  title: "Avatar"
};

export const Stories = () => (
  <>
    <Avatar label={"N"}/>
    <br/>
    <Avatar label={"NI"}/>
  </>
);
