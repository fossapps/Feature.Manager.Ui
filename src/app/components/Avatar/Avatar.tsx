import React from "react";
import { style } from "typestyle";
import { BorderRadius, Color, Spacing } from "../../constants";

interface IProp {
  label: string;
}
const avatarStyle = style({
  alignItems: "center",
  backgroundColor: Color.PRIMARY,
  borderRadius: BorderRadius.ROUND,
  color: Color.WHITE,
  display: "flex",
  fontSize: "1.25rem",
  height: Spacing.XXL,
  justifyContent: "center",
  padding: Spacing.S,
  width: Spacing.XXL
});
export const Avatar: React.FC<IProp> = (prop) => {
  return (<div className={avatarStyle}>{prop.label.split("")[0]}</div>);
};
