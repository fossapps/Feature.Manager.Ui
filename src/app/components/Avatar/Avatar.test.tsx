import { shallow } from "enzyme";
import React from "react";
import { Avatar } from "./Avatar";

describe("Avatar", () => {
  it("should match snapshot", () => {
    expect(shallow(<Avatar label={"N"}/>)).toMatchSnapshot();
    expect(shallow(<Avatar label={"NISHCHAL"}/>)).toMatchSnapshot();
  });
});
