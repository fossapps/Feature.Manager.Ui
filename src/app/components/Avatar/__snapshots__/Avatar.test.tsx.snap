// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<div
  className="fih5kfy"
>
  N
</div>
`;

exports[`Avatar should match snapshot 2`] = `
<div
  className="fih5kfy"
>
  N
</div>
`;
